cowbird/services/impl/magpie.py

```diff
@@ -20,20 +32,76 @@ class Magpie(Service):
     required_params = [SERVICE_URL_PARAM]
 
     def __init__(self, settings, name, **kwargs):
-        # type: (SettingsType, str, dict) -> None
+        # type: (SettingsType, str, Dict) -> None
         """
         Create the magpie instance and instantiate the permission synchronizer that will handle the permission events.
 
-        @param settings: Cowbird settings for convenience
-        @param name: Service name
+        :param settings: Cowbird settings for convenience
+        :param name: Service name
         """
         super(Magpie, self).__init__(settings, name, **kwargs)
+
+        self.headers = {"Content-type": CONTENT_TYPE_JSON}
+        self.admin_user = kwargs.get(MAGPIE_ADMIN_USER_TAG, None)
+        self.admin_password = kwargs.get(MAGPIE_ADMIN_PASSWORD_TAG, None)
```

Comment:
The reason those are in the `kwargs` is because of how services (handlers) are created by the `ServiceFactory` : 
https://github.com/Ouranosinc/cowbird/blob/643e0740558665c2486ad410ef05ff9ece24bab7/cowbird/services/service_factory.py#L52

All cowbird services (handlers) are created by the `ServiceFactory`, where the `kwargs` contain a variable number of keyword arguments taken from the related service's config. Here Magpie has 2 args that the other handlers don't have, but if I would want them in the signature directly, I would have to add a special case in the `ServiceFactory`, so I think considering the implementation, it makes more sense to keep it as it is.

Reply:
You can use keywords like so: `__init__(settings, name, admin_user=None, admin_password=None, **kwargs)`. 
They will be mapped automatically without need to do `kwargs.get`. 
It also makes it more explicit of what is expected as input (since keywords must be passed anyway) without having to dig in the code.

Reply:
You could even omit the `**kwargs` to make it explicit that anything else is unhandled. This would actually raise an exception if something invalid was passed, but I don't mind as much to just ignore the extra arguments also.